posthog/models/sharing_configuration.py

```diff
@@ -19,3 +20,15 @@ class SharingConfiguration(models.Model):
     access_token: models.CharField = models.CharField(
         max_length=400, null=True, blank=True, default=get_default_access_token, unique=True
     )
+
+    def get_connected_insight_ids(self) -> List[int]:
+        if self.insight:
+            return [self.insight.id]
+        elif self.dashboard:
+            # Check whether this sharing configuration's dashboard contains this insight
+            return list(
+                self.dashboard.tiles.exclude(deleted=True, insight__isnull=True, insight__deleted=True).values_list(
```

Comment:
Do you want to exclude `deleted=True AND insight__isnull=True AND insight__deleted=True` or the same with `OR`-s? [According to Django](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#exclude) it'll do the `AND`-s, though it looks like we'd want to exclude insights when any of these is true?

Reply:
Good catch! Now, I just added tests to make sure that insight with any link in the chain deleted (insight, tile, dashboard) cannot be retrieved with the sharing access token _after fixing this_, but it turns out this already is guaranteed by all the implicit `objects` vs. `objects_including_soft_deleted` filtering that @pauldambra added – so we're safe. (Nevertheless added a few test cases + cleaned up some handling of `.deleted` model instances.)